src/coreclr/gc/gc.cpp

```diff
@@ -49307,3 +49377,294 @@ void PopulateDacVars(GcDacVars *gcDacVars)
+            for (int i = 0; i < count_free_region_kinds; i++)
+            {
+                committed_free += heap->free_regions [i].get_size_committed_in_free();
+                heap_segment* seg = heap->free_regions [i].get_first_free_region();
+                while (seg != nullptr)
+                {
+                    committed_bookkeeping += heap->get_mark_array_size (seg);
+                    seg = seg->next;
+                }
+            }
+        }
+        for (int i = 0; i < count_free_region_kinds; i++)
+        {
+            committed_free += global_regions_to_decommit [i].get_size_committed_in_free();
+            heap_segment* seg = global_regions_to_decommit [i].get_first_free_region();
+            while (seg != nullptr)
+            {
+#ifdef MULTIPLE_HEAPS
+                gc_heap* heap = g_heaps [0];
+#else
+                gc_heap* heap = nullptr;
+#endif //MULTIPLE_HEAPS
+                committed_bookkeeping += heap->get_mark_array_size (seg);
+                seg = seg->next;
+            }
+        }
+        committed_free += global_free_huge_regions.get_size_committed_in_free();
+        {
+            heap_segment* seg = global_free_huge_regions.get_first_free_region();
+            while (seg != nullptr)
+            {
+#ifdef MULTIPLE_HEAPS
+                gc_heap* heap = g_heaps [0];
+#else
+                gc_heap* heap = nullptr;
+#endif //MULTIPLE_HEAPS
+                committed_bookkeeping += heap->get_mark_array_size (seg);
+                seg = seg->next;
+            }
+        }
+
+        committed_by_oh [recorded_committed_free_bucket] = committed_free;
+        total_committed += committed_free;
+
+        // Accounting for the bytes committed for the book keeping elements
+        uint8_t* commit_begins[total_bookkeeping_elements];
+        size_t commit_sizes[total_bookkeeping_elements];
+        size_t new_sizes[total_bookkeeping_elements];
+        bool get_card_table_commit_layout_result = get_card_table_commit_layout(g_gc_lowest_address, bookkeeping_covered_committed, commit_begins, commit_sizes, new_sizes);
+        assert (get_card_table_commit_layout_result);
+
+        for (int i = card_table_element; i <= seg_mapping_table_element; i++)
+        {
+            // In case background GC is disabled - the software write watch table is still there
+            // but with size 0
+            assert (commit_sizes [i] >= 0);
+            committed_bookkeeping += commit_sizes [i];
+        }
+        current_total_committed_bookkeeping = committed_bookkeeping;
+        committed_by_oh [recorded_committed_bookkeeping_bucket] = committed_bookkeeping;
+        total_committed += committed_bookkeeping;
+
+        current_total_committed = total_committed;
+#endif //USE_REGIONS
+#ifdef COMMITTED_BYTES_SHADOW
+        assert (current_total_committed == current_total_committed_shadow);
+        assert (current_total_committed_bookkeeping == current_total_committed_bookkeeping_shadow);
```

Comment:
seems like we might as well also validate the shadow values for per object heap also match up with the ones you just computed.